pkg/console/validator.go

```diff
@@ -39,23 +39,33 @@ func prettyError(errMsg string, value string) error {
 	return errors.Errorf("%s: %s", errMsg, value)
 }
 
-func checkInterface(name string) error {
-	if name == "" {
+func checkInterface(iface config.NetworkInterface) error {
+	// For now we only accept interface name but not device specifier.
+	if iface.Name == "" {
 		return errors.New(ErrMsgInterfaceNotSpecified)
 	}
 	ifaces, err := net.Interfaces()
 	if err != nil {
 		return err
 	}
+
+	checkFlags := func(flags net.Flags, name string) error {
+		if flags&net.FlagLoopback != 0 {
+			return prettyError(ErrMsgInterfaceIsLoop, name)
+		}
+		return nil
+	}
+
 	for _, i := range ifaces {
-		if i.Name == name {
-			if i.Flags&net.FlagLoopback != 0 {
-				return prettyError(ErrMsgInterfaceIsLoop, name)
-			}
-			return nil
+		if i.Name == iface.Name {
+			return checkFlags(i.Flags, iface.Name)
 		}
 	}
-	return prettyError(ErrMsgInterfaceNotFound, name)
+
+	if iface.Name == "" {
+		return prettyError(ErrMsgInterfaceNotFound, iface.Name)
+	}
+	return prettyError(ErrMsgInterfaceNotFound, iface.HwAddr)
```

Comment:
I'd suggest also including the interface name here, HwAddr is not implemented yet.